tests/utils.py

```diff
@@ -146,3 +151,71 @@ def set_request_limit(client, limit):
     for limit_name, limit_val in tmp.items():
         if hasattr(client, limit_name):
             setattr(client, limit_name, limit_val)
+
+
+def cdf_aggregate(
+    raw_df: pandas.DataFrame,
+    aggregate: Literal["average", "sum", "count"],
+    granularity: str,
+    is_step: bool = False,
+    raw_freq: str = None,
+) -> pandas.DataFrame:
+    """Aggregates the dataframe as CDF is doing it on the database layer.
+
+    **Motivation**: This is used in testing to verify that the correct aggregation is done with
+    on the client side when aggregating in given time zone.
+
+    Current assumptions:
+        * No step timeseries
+        * Uniform index.
+        * Known frequency of raw data.
+
+    Args:
+        raw_df (pd.DataFrame): Dataframe with the raw datapoints.
+        aggregate (str): Single aggregate to calculate, supported average, sum.
+        granularity (str): The granularity to aggregates at. e.g. '15s', '2h', '10d'.
+        is_step (bool): Whether to use stepwise or continuous interpolation.
+        raw_freq (str): The frequency of the raw data. If it is not given, it is attempted inferred from raw_df.
+    """
+    if is_step:
+        raise NotImplementedError()
+
+    pd = cast(Any, local_import("pandas"))
+    granularity_pd = granularity.replace("m", "T")
+    grouping = raw_df.groupby(pd.Grouper(freq=granularity_pd))
+    if aggregate == "sum":
+        return grouping.sum()
+    elif aggregate == "count":
+        return grouping.count().astype("Int64")
+
+    # The average is calculated by the formula '1/(b-a) int_a^b f(t) dt' where f(t) is the continuous function
+    # This is weighted average of the sampled version of f(t)
+    np = cast(Any, local_import("numpy"))
+
+    def integrate_average(values: pandas.DataFrame) -> pandas.Series:
+        return pd.Series(((values.iloc[1:].values + values.iloc[:-1].values) / 2.0).mean(), index=values.columns)
```

Comment:
Avoids runtimewarning from numpy:
```suggestion
    def integrate_average(values: pandas.DataFrame) -> pandas.Series:
        inner = ((values.iloc[1:].values + values.iloc[:-1].values) / 2.0)
        res = inner.mean() if inner.shape[0] else np.nan
        return pd.Series(res, index=values.columns)
```